src/dbt_client/dbtCloudIntegration.ts

```diff
@@ -1023,6 +1023,15 @@ export class DBTCloudProjectIntegration
     }
   }
 
+  private firstRun = false;
+  private async retryOnce() {
+    if (this.firstRun) {
+      return;
+    }
+    this.firstRun = true;
+    this.findAdapterType();
```

Comment:
@mdesmet, should we add a telemetry event for retry and maybe for failure as well?